webapp/components/status_dropdown.jsx

```diff
@@ -0,0 +1,137 @@
+import PreferenceStore from 'stores/preference_store.jsx';
+
+const STATUSES = ['online', 'away', 'offline'];
+
+export default class StatusDropdown extends React.Component {
+
+    static propTypes = {
+        status: React.PropTypes.string
+    }
+
+    constructor(props) {
+        super(props);
+
+        [
+            'toggleDropdown',
+            'setStatus',
+            'renderStatusAction',
+            'onMouseEnter',
+            'onMouseLeave'
+        ].forEach((fnName) => {
+            this[fnName] = this[fnName].bind(this);
+        });
+
+        this.state = {
+            showDropdown: false,
+            mouseOver: false
+        };
+    }
+
+    onMouseEnter() {
+        this.setState({mouseOver: true});
+    }
+
+    onMouseLeave() {
+        this.setState({mouseOver: false});
+    }
+
+    renderStatusIcon() {
+        const team = TeamStore.getCurrent();
+        const theme = PreferenceStore.getTheme(team.id);
+
+        if (this.state.mouseOver) {
+            return (
+                <span
+                    className={'status'}
+                    style={{color: theme.sidebarHeaderTextColor}}
+                >
+                    <i
+                        className={'fa fa-edit'}
+                        style={{padding: '1px 2px'}}
+                    />
+                </span>
+            );
+        }
+        return (
+            <StatusIcon
+                status={this.props.status}
+            />
+        );
+    }
+
+    toggleDropdown() {
+        const newState = {showDropdown: !this.state.showDropdown};
+        this.setState(newState);
+    }
+
+    setStatus = (status) => (event) => {
+        event.preventDefault();
+
+        if (STATUSES.indexOf(status) < 0) {
+            throw new Error(`Invalid status command: ${status}`);
+        }
+
+        const channel = ChannelStore.getCurrent();
+        const channelId = channel.id;
+        const args = {channel_id: channelId};
+
+        ChannelActions.executeCommand(
+            `/${status}`,
+            args,
+            null,
+            null
+        );
+    }
+
+    renderStatusAction(status) {
+        const onClick = this.setStatus(status);
+        return (
+            <li key={status}>
+                <a
+                    href={'#'}
+                    onClick={onClick}
+                >
+                    <FormattedMessage
+                        id={`status_dropdown.set_${status}`}
+                        defaultMessage={status}
+                    />
+                </a>
+            </li>
+        );
+    }
+
+    render() {
+        const {showDropdown} = this.state;
+        const statusIcon = this.renderStatusIcon();
+        const actions = STATUSES.map(this.renderStatusAction);
+        return (
+            <Dropdown
+                id={'status-dropdown'}
+                defaultOpen={showDropdown}
+            >
+                <span
+                    bsRole={'toggle'}
```

Comment:
It's kind of silly, but we need to use another component here instead of a raw span, even if that component just renders the span. Otherwise, React will constantly print warnings about us passing unrecognized bootstrap props (`bsRole` and `bsClass`) into a span.

We've hit the same warning in other places, so could you maybe make a generic `BootstrapSpan` component that just renders a span, but also swallows the invalid props? Something small that we can use elsewhere like
```
class BootstrapSpan extends React.PureComponent {
    render() {
        let {children, ...props} = this.props;
        delete props.bsRole;
        delete props.bsClass;

        return <span {...props}>{children}</span>;
    }
}
```